elyra/pipeline/pipeline.py

```diff
@@ -141,20 +141,19 @@ def env_vars_as_dict(self, logger: Optional[Logger] = None) -> Dict:
         """
         Operation stores environment variables in a list of name=value pairs, while
         subprocess.run() requires a dictionary - so we must convert.  If no envs are
-        configured on the Operation, the existing env is returned, otherwise envs
-        configured on the Operation are overlayed on the existing env.
+        configured on the Operation, an empty dictionary is returned, otherwise envs
+        configured on the Operation are converted to dictionary entries and returned.
         """
         envs = {}
         for nv in self.env_vars:
             if nv and len(nv) > 0:
                 nv_pair = nv.split("=")
                 if len(nv_pair) == 2 and nv_pair[0].strip():
                     envs[nv_pair[0]] = nv_pair[1]
+                elif logger:
```

Comment:
Follow up to this scenario:
 - User sets variable to `MYVAR=3=2`.
 - VPE does not raise a validation error
 - A warning message is logged and the variable ignored but the user never explicitly notified.

```
Could not process environment variable entry `MYVAR=3=2`, skipping...
```

Would it be better to 
- treat this as fatal error on the back-end
- add code to the validation engine to alert the user?

Reply:
"add code to the validation engine to alert the user?" -> new issue

Reply:
I think we should preserve the intention of the user and not treat any inability to parse environment variables as fatal at this particular location.  Should we treat these as errors detected during validation?  Perhaps, but that's not in the scope of this issue.

Regarding `MYVAR=` scenarios.  This was discussed in today's scrum and the consensus was that this particular _non-value_ scenario should not result in `MYVAR`'s inclusion in the environment of the runtime.  However, an explicit, yet empty, scenario like `MYVAR=""` _should_ result in `MYVAR`'s inclusion in the environment.

If you're in agreement here, I can include this functionality in this PR - please advise.

Reply:
> If you're in agreement here, 

Yes, that makes sense

How about handling multiple `=` signs in the backend code? https://github.com/elyra-ai/elyra/pull/1701#discussion_r639863979. Having thought about this some more I believe `A=2=3` could be a valid scenario and should not be ignored. I am thinking about the component library here, which uses environment variables extensively and might have a need to have an `=` sign somewhere in the value.